### PR TITLE
support nbd on euler or arm

### PR DESCRIPTION
**Pull request by muxuelanKK**
# Describe what this PR does #

For euler or arm，command  “rbd device list --format=json --device-type=nbd” can be response slowly, often more than 5 miniutes. So  we use rbd-nbd list-mapped instead.

## Is there anything that requires special attention ##

Do you have any questions?

Is the change backward compatible?

Are there concerns around backward compatibility?

Provide any external context for the change, if any.

### CODE REVIEW

**Comment by nixpanic** [comment]
I think this looks reasonable. Could you update the title of the commit to

    rbd: support nbd on euler or arm

so that **commitlint** will accept it? You'll need to force-push that change to your branch that you used to create this PR from.

Thanks!

File: internal/rbd/rbd_attach.go [hunk 1]
```diff
@@ -352,29 +355,27 @@ func attachRBDImage(ctx context.Context, volOptions *rbdVolume, device string, c
 }
 
 func appendNbdDeviceTypeAndOptions(cmdArgs []string, userOptions, cookie string) []string {
-	cmdArgs = append(cmdArgs, "--device-type", accessTypeNbd)
-
 	isUnmap := CheckSliceContains(cmdArgs, "unmap")
 	if !isUnmap {
 		if !strings.Contains(userOptions, useNbdNetlink) {
-			cmdArgs = append(cmdArgs, "--options", useNbdNetlink)
+			cmdArgs = append(cmdArgs, "--"+useNbdNetlink)
```
Comment (nixpanic) [inline internal/rbd/rbd_attach.go:370]: My main concern with this patch is that the `--options` parameter seems to be dropped. Is that intentional, and does it keep working with the currently maintained Ceph versions?

**Comment by nixpanic** [comment]
Inspired by this PR, I've created #4364 which will cause a little conflict, sorry!

**Comment by nixpanic** [comment]
> /test ci/centos/mini-e2e/k8s-1.29

This passed, so I think all still works. Ideally it is more efficient now too.

@pkalever can you have a quick look and check if it doesn't break anything?

**Review by nixpanic — APPROVED** [review]
CI job passed, so functionality should be OK.

**Review by Madhu-1 — APPROVED** [review]

**Comment by nixpanic** [comment]
@mergifyio rebase

**Comment by nixpanic** [comment]
@mergifyio queue

**Comment by nixpanic** [comment]
@mergifyio rebase

**Comment by nixpanic** [comment]
@mergifyio queue

**Pull request merged**